YOLayout/YOLayout.m

```diff
@@ -65,7 +65,17 @@ - (CGSize)sizeThatFits:(CGSize)size {
 }
 
 - (CGRect)setFrame:(CGRect)frame view:(id)view sizeToFit:(BOOL)sizeToFit {
-  return [self setFrame:frame view:view options:(sizeToFit ? YOLayoutOptionsSizeToFit : 0)];
+
+  YOLayoutOptions options = 0;
+  // If UILabel and 1 line, we need to fix size to fit options to constrain width,
+  // since UILabel will return a size > the width we specify.
+  if (sizeToFit && [view isKindOfClass:UILabel.class] && ((UILabel *)view).numberOfLines == 1) {
```

Comment:
I don't like special casing for individual view classes. There are other views that size horizontally as well, such as `UIButton` & `UITextField` (& I think `UIImageView` if `image != nil`). There's also nothing stopping a user's YOView subclass from returning a size greater than the passed in size. Seems like a lot of opportunity for inconsistency when including special cases for specific views.

I'm more open to making `YOLayoutOptionsConstrainWidth` the default for `sizeToFit:YES`. My hesitation is that its inconsistent in the behavior for sizing views horizontally and vertically. For example, IIRC UIImageView doesn't constrain it's width so you can easily size it based on it's image's size. Currently you can do:

    [layout setFrame:CGRectMake(10, 10, 0, 0) view:yoself.imageView sizeToFit:YES];

With this change it'd have to be something like.

    [layout setFrame:CGRectMake(10, 10, yoself.imageView.image.size.width, 0) view:yoself.imageView sizeToFit:YES];

More philosophically, you could also imagine an interface that scrolls infinitely horizontally. You might have elements that could grow arbitrarily horizontally wide. I don't like the idea of requiring people to pass in `CGFLOAT_MAX` as the width. Though, admittedly, this is a very hypothetical use case.

A third option would be to rename this method `setFrame:view:sizeToFitVertical:` or even `setFrame:sizeVerticallyForView:` or something similar. We then only use `YOLayoutOptionsSizeToFitVertical`, and leave the width fixed. I'd feel good about this since it's more descriptive about what's actually happening.

Thoughts?